sweep/fee_bumper.go

```diff
@@ -0,0 +1,623 @@
+	}
+}
+
+// NotifyConfirm tells the bumper a transaction has been confirmed and needs to
+// be removed from the monitor queue.
+//
+// NOTE: part of the Bumper interface.
+func (f *feeBumper) NotifyConfirm(op *wire.OutPoint) {
+	input, ok := f.monitoredInputs.LoadAndDelete(op)
+	if !ok {
+		feeLog.Infof("Notify the spend of outpoint: %v not found in "+
+			"monitored txes, might have been deleted", op)
+		return
+	}
+
+	feeLog.Infof("Notified the spent: outpoint=%v deleted", op)
+
+	feeLog.Tracef("Outpoint confirmed: %s", input)
+}
+
+// validateRequest checks that the request has sane params used.
+func (f *feeBumper) validateRequest(req *MonitorRequest) error {
+	// Validate the strategy used.
+	if err := validateStrategy(req.Strategy); err != nil {
+		return err
+	}
+
+	// Validate conf target and deadline height.
+	if err := validateConfTarget(req); err != nil {
+		return err
+	}
+
+	feeLog.Debugf("Validated request for outpoint: %v",
+		req.Input.OutPoint())
+
+	return nil
+}
+
+// validateConfTarget validates the conf target is sane.
+func validateConfTarget(req *MonitorRequest) error {
+	// Check conf target is greater than 0.
+	if req.ConfTarget < 1 {
+		return ErrConfTargetTooSmall
+	}
+
+	if req.ConfTarget == NoHeightInfo && req.FeeRate == 0 {
+		return fmt.Errorf("%w: must set conf target or fee rate",
+			ErrConflictArgs)
+
+	}
+
+	return nil
+}
+
+// RequestMonitor registers an outpoint to be monitored for potential fee
+// bumping. If the outpoint already exits, ErrAlreadyMonitored will be
+// returned.
+//
+// NOTE: part of the Bumper interface.
+func (f *feeBumper) RequestMonitor(req *MonitorRequest) error {
+	feeLog.Debugf("Received request %s", req)
+
+	op := req.Input.OutPoint()
+
+	// Check if the request has already been processed.
+	//
+	// TODO(yy): allow overwrite?
+	input, loaded := f.monitoredInputs.Load(op)
+	if loaded {
+		feeLog.Errorf("Already monitored for %s", input)
+
+		return ErrAlreadyMonitored
+	}
+
+	// Validate the request params.
+	err := f.validateRequest(req)
+	if err != nil {
+		return fmt.Errorf("invalid record: %w", err)
+	}
+
+	// If the request doesn't specify a fee rate used, we need to get the
+	// current fee rate from our fee estimator.
+	feeRateUsed := req.FeeRate
+	if feeRateUsed == 0 {
+		feeRate, err := f.estimator.EstimateFeePerKW(req.ConfTarget)
+		if err != nil {
+			return fmt.Errorf("failed to estimate fee: %w", err)
+		}
+
+		// Save the fee rate used for this request.
+		feeRateUsed = feeRate
+	}
+
+	// If the requst doesn't specify a broadcast height, use the manager's
+	// best known height.
+	broadcastHeight := req.Height
+	if broadcastHeight == 0 {
+		// Save the current height used for this request.
+		broadcastHeight = f.currentHeight
+	}
+
+	// Calculate the deadline height based on the broadcast height and the
+	// conf target.
+	var deadlineHeight uint32
+
+	// Calculate the deadline height if the conf target is set.
+	if req.ConfTarget != NoHeightInfo {
```

Comment:
Shouldn't this be `==`? Since in this case, no conf target was specified, so we need to enforce a default conf target.